### DISCUSSION TITLE
nvme spams proxmox syslog

### DISCUSSION THREAD

**Discussion by cholzer79** [netdata/netdata#16002] [Q&A]
I am running netdata on a Proxmox VE 8 machine. 

PVE is build upon debian and does not have sudo out of the box. So for netdata/nvme I was forced to install sudo and set it up.

The netdata nvme integration works, however every 10 seconds it spams my syslog with this.

`Sep 18 20:08:07 pve sudo[102109]:  netdata : PWD=/etc/netdata ; USER=root ; COMMAND=/usr/sbin/nvme smart-log /dev/nvme3n1 --output-format=json
Sep 18 20:08:07 pve sudo[102109]: pam_unix(sudo:session): session opened for user root(uid=0) by (uid=109)
Sep 18 20:08:07 pve sudo[102109]: pam_unix(sudo:session): session closed for user root
Sep 18 20:08:07 pve sudo[102114]:  netdata : PWD=/etc/netdata ; USER=root ; COMMAND=/usr/sbin/nvme smart-log /dev/nvme2n1 --output-format=json
Sep 18 20:08:07 pve sudo[102114]: pam_unix(sudo:session): session opened for user root(uid=0) by (uid=109)
Sep 18 20:08:07 pve sudo[102114]: pam_unix(sudo:session): session closed for user root
Sep 18 20:08:07 pve sudo[102116]:  netdata : PWD=/etc/netdata ; USER=root ; COMMAND=/usr/sbin/nvme smart-log /dev/nvme1n1 --output-format=json
Sep 18 20:08:07 pve sudo[102116]: pam_unix(sudo:session): session opened for user root(uid=0) by (uid=109)
Sep 18 20:08:07 pve sudo[102116]: pam_unix(sudo:session): session closed for user root
Sep 18 20:08:07 pve sudo[102118]:  netdata : PWD=/etc/netdata ; USER=root ; COMMAND=/usr/sbin/nvme smart-log /dev/nvme0n1 --output-format=json
Sep 18 20:08:07 pve sudo[102118]: pam_unix(sudo:session): session opened for user root(uid=0) by (uid=109)
Sep 18 20:08:07 pve sudo[102118]: pam_unix(sudo:session): session closed for user root
Sep 18 20:08:07 pve sudo[102120]:  netdata : PWD=/etc/netdata ; USER=root ; COMMAND=/usr/sbin/nvme smart-log /dev/nvme4n1 --output-format=json
Sep 18 20:08:07 pve sudo[102120]: pam_unix(sudo:session): session opened for user root(uid=0) by (uid=109)
Sep 18 20:08:07 pve sudo[102120]: pam_unix(sudo:session): session closed for user root`

Is there any way around that?

**Comment by ilyam8** [comment]
Hey, @cholzer79. There are a few options:

 1. disable nvme collector.
 2. disable logging successful sudo attempts. See [man sudo logging](https://www.sudo.ws/docs/man/1.9.2/sudoers.man/#Logging). 

For 2 add the following to the sudoers file:

```
Defaults        !log_allowed
```

**Reply by ilyam8** [reply]
An alternative solution is to disable logging for `/usr/sbin/nvme`. Quick googling: can be achieved using [Cmnd_Alias](https://www.sudo.ws/docs/man/1.9.2/sudoers.man/#Aliases). Something like this should do (I didn't test it):

```
Cmnd_Alias    NVME_SCRIPT = /usr/sbin/nvme
Defaults     !NVME_SCRIPT
```

**Comment by cholzer79** [comment]
> Hey, @cholzer79. There are a few options:
> 
> 1. disable nvme collector.
> 2. disable logging successful sudo attempts. See [man sudo logging](https://www.sudo.ws/docs/man/1.9.2/sudoers.man/#Logging).
> 
> For 2 add the following to the sudoers file:
> 
> ```
> Defaults        !log_allowed
> ```
Thanks! You got me on the right track.

The soludtion to disable all logging related to the nvme integration (which I do want to use) was to add this line to /etc/sudoers
```
Defaults:netdata !syslog, !pam_session
```

So /etc/sudoers looks like this now:
```
netdata ALL=(root) NOPASSWD: /usr/sbin/nvme
Defaults:netdata !syslog, !pam_session
```
